app/controllers/presentations_controller.rb

```diff
@@ -1,8 +1,10 @@
 class PresentationsController < ApplicationController
   def index
-    @presentations = Presentation.joins(:event)
-                                 .merge(Event.order(date: :desc))
-                                 .page(params[:page])
+    @q = Presentation.ransack(params[:q])
```

Comment:
It's a ransack default search parameter. You can customize it, but I left the default for the time being to minimize query string length.
https://github.com/activerecord-hackery/ransack/tree/v1.8.2#simple-mode